fem/kernels.hpp

```diff
@@ -48,6 +48,26 @@ MFEM_HOST_DEVICE inline void LoadB(const int D1D, const int Q1D,
    MFEM_SYNC_THREAD;
 }
 
+/// Load B1d matrice into shared memory
+MFEM_HOST_DEVICE inline void LoadB(const int D1D, const int Q1D,
+                                   const ConstDeviceMatrix &b,
+                                   const DeviceMatrix &B)
```

Comment:
I agree - the `DeviceTensor` class is for now a wrapper around a pointer which can be changed or not, the underlying `data` is not what the first `const` is referring to.

Reply:
This is normal for "view"-like objects, see e.g. C++20's `std::span` -- the const-ness applies to the `span` object itself, not to the data it is referring to. The same is true for the proposed multi-dimensional version, `mdspan`, see https://github.com/kokkos/mdspan. Perhaps, using a different name (instead of `DeviceTensor` and `DeviceMatrix`) could have been a better choice -- maybe we can address that as part of #2626.

Reply:
@camierjs and @v-dobrev I guess while this might be the "norm" with c++ view objects it still doesn't make me feel any safer seeing this in practice as it just seems like an easy way to shot yourself in the foot...

As I'm sure nothing can be really be done in the short term (would #2626 allow for stronger safety guarantees here?), could we at least for the cases here add in a note which one of the const "views" -objects are being mutated so users/developers be able to easily scan the docs and know what is allowed to be modified.

Reply:
I agree: pointing this out in the doxygen comments will be helpful.

Reply:
#2626 indeed resolves this issue.

Reply:
Will add the documentation.